pytket/extensions/azure/backends/azure.py

```diff
@@ -177,18 +197,33 @@ def process_circuits(
 
         handles = []
         for i, (c, n_shots) in enumerate(zip(circuits, n_shots_list)):
-            qkc = tk_to_qiskit(c)
-            module, entry_points = to_qir_module(qkc)
-            assert len(entry_points) == 1
             input_params = {
-                "entryPoint": entry_points[0],
+                "entryPoint": "main",
                 "arguments": [],
                 "count": n_shots,
             }
+            if self._backendinfo.device_name == "ionq.simulator":
+                module_bitcode = pytket_to_qir(
+                    c,
+                    qir_format=QIRFormat.BINARY,
+                    int_type=64,
+                    cut_pytket_register=False,
+                    profile=QIRProfile.AZUREBASE,
+                )
+                raise ValueError("ionq devices currently not supported")
```

Comment:
I had removed this, because this had issues with a lot of gates and it is unclear what is supported  / should be supported. I have added this now back in with the same support as we had before. This is probably fine, as this is all not a state to use in production.